packages/fuels-signers/src/wallet.rs

```diff
@@ -679,15 +682,27 @@ impl WalletUnlocked {
             .map(|resource| resource.amount())
             .sum();
 
-        let predicate_data = data.unwrap_or_default();
+        // Iterate through the spendable resources and calculate the appropriate offsets
+        // for the coin or message predicates
+        let mut offset = base_predicate_offset(&predicate.consensus_parameters().await?);
         let inputs = spendable_predicate_resources
             .into_iter()
             .map(|resource| match resource {
                 Resource::Coin(coin) => {
-                    self.create_coin_predicate(coin, asset_id, code.clone(), predicate_data.clone())
+                    offset += coin_predicate_data_offset(code.len());
```

Comment:
Thinking if this shouldn't be unit tested. Quite a lot happening. 

To e2e test this you'd have to have multiple coins and messages tied to the predicate and have the predicate accept vectors as data and then spend the resources from the predicate.

Is there such a test? I know vectors are tested but wasn't sure how many inputs were used -- whether the offset is ever anything other than `base_predicate_offset`?

Reply:
We have exactly that test. And all the other test also include multiple messages and coins. I thought about moving this logic somewhere else but then got stuck and could not decide where to put it so I left it there..

The first offset is always `base_predicate_offset` (of course if it is not changed in the VM) then we have a list of inputs and we have to iterate through them and add the right offset to point to the predicate data part of the specific input